sopel/irc.py

```diff
@@ -211,25 +216,45 @@ def handle_connect(self):
                                            suppress_ragged_eofs=True,
                                            cert_reqs=ssl.CERT_REQUIRED,
                                            ca_certs=self.ca_certs)
+                # connect to host specified in config first
                 try:
                     ssl.match_hostname(self.ssl.getpeercert(), self.config.core.host)
                 except ssl.CertificateError:
-                    stderr("Invalid certficate, hostname mismatch!")
-                    os.unlink(self.config.core.pid_file_path)
-                    os._exit(1)
+                    # the host in config and certificate don't match
+                    LOGGER.error("hostname mismatch between configuration and certificate")
+                    # check (via exception) if a CNAME matches as a fallback
+                    has_matched = False
+                    for hostname in self._get_cnames(self.config.core.host):
+                        try:
+                            ssl.match_hostname(self.ssl.getpeercert(), hostname)
+                            LOGGER.warning("using {0} instead of {1} for TLS connection".format(hostname,
+                                                                                             self.config.core.host))
```

Comment:
Along with the typos below, please move the `.format(hostname,` to the next line and align it with the containing function call, so this looks less weird.

````
                            LOGGER.warning("using {0} instead of {1} for TLS connection"
                                            .format(hostname, self.config.core.host))
````